lib/src/entity/entity_manager.dart

```diff
@@ -164,6 +177,8 @@ class EntityManager {
     if (_entitiesByName.containsKey(entity.name)) {
       _entitiesByName.remove(entity.name);
     }
-    entity._pool?.release(entity);
+    entity.dispose();
+    //TODO: Why?
+    //entity._pool?.release(entity);
```

Comment:
Because of the above comment on `dispose`, that is the reason why we release the entity back into the pool, so that it can be reused later on. Saving precious initialisation time.